perceval/backends/gerrit.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Alvaro del Castillo San Felix <acs@bitergia.com>
+#
+
+'''Gerrit backend for Perseval'''
+
+
+from datetime import datetime
+from dateutil import parser
+import json
+import logging
+import os.path
+import re
+import subprocess
+from time import time
+
+from ..backend import Backend, BackendCommand
+from ..cache import Cache
+from ..errors import BackendError, CacheError
+from ..utils import DEFAULT_DATETIME, str_to_datetime
+
+
+
+class Gerrit(Backend):
+
+    name = "gerrit"
+
+    def __init__(self, user = None, url = None, nreviews = None,
+                 cache = None, **nouse):
+        super().__init__(cache=cache)
+        self.repository = url
+        self.nreviews = nreviews
+        self.last_item = None  # Start item for next iteration
+        self.more_updates = True  # To check if reviews are updates
+        self.number_results = self.nreviews  # To control if there are more items
+        self.client = GerritClient(self.repository, user, nreviews)
+
+
```

Comment:
This semantic should be in some place: howto get the id for the repository and howto get the unique id for the items. But probably it should be outside Perceval as you propose. I will rework the integration with p2o to understand the best place to put this logic/semantic in the toolchain.